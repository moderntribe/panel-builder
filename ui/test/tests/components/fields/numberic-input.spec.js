/* global expect b:true */

import React from 'react';
import { shallow, mount } from 'enzyme';
import NumericInput from 'components/fields/numeric-input';
import renderer from 'react-test-renderer';

const props = {
	name: 'padding-top',
	label: 'Top',
	default: 0,
	step: 2,
	max: 3000,
	layout: 'compact',
	input_width: 3,
};

describe('Numeric input field', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<NumericInput {...props} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders as a <div>', () => {
		const wrapper = shallow(<NumericInput step={1} />);
		expect(wrapper.type()).toEqual('div');
	});

	it('allows us to set props', () => {
		const wrapper = mount(<NumericInput step={1} description="This is a test" />);
		expect(wrapper.props().description).toEqual('This is a test');
		wrapper.setProps({ description: 'Change props for test' });
		expect(wrapper.props().description).toEqual('Change props for test');
	});

	it('has a label', () => {
		const wrapper = shallow(<NumericInput step={1} />);
		expect(wrapper.find('label').length).toEqual(1);
	});
});
