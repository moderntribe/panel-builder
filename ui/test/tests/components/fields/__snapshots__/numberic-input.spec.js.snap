exports[`Numeric input field renders correctly 1`] = `
<div
  className="field compact panel-field">
  <span
    className="container">
    <span
      className="react-numeric-input"
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={null}>
      <input
        className="input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        style={null}
        type="text"
        value="0" />
      <b
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={null}>
        <i
          style={null} />
      </b>
      <b
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={null}>
        <i
          style={null} />
      </b>
    </span>
  </span>
  <label
    className="label panel-field-label">
    Top
  </label>
</div>
`;
